applications/crossbar/src/modules/cb_search.erl

```diff
@@ -125,27 +165,104 @@ validate_search(Context, Type, Query) ->
+validate_multi(Context, 'undefined') ->
+    cb_context:add_validation_error(
+        <<"t">>
+        ,<<"required">>
+        ,wh_json:from_list([{<<"message">>, <<"Search needs a document type to search on">>}])
+        ,Context
+    );
+validate_multi(Context, Type) ->
+    QS = cb_context:query_string(Context),
+    case wh_json:to_proplist(wh_json:delete_key(<<"t">>, QS)) of
+        [_|_]=Props -> validate_multi(Context, Type, Props);
+        _Other ->
+            cb_context:add_validation_error(
+                <<"multi">>
+                ,<<"enum">>
+                ,wh_json:from_list([
+                    {<<"message">>, <<"Multi Search needs something to search like a doc type">>}
+                    ,{<<"target">>, ?SEARCHABLE}
+                 ])
+                ,Context
+            )
+    end.
+
+validate_multi(Context, Type, Props) ->
+    QueryOptions = query_options(cb_context:account_db(Context)),
+    Context1 = validate_query(Context, QueryOptions, Props),
+    case cb_context:resp_status(Context1) of
+        'success' ->
+            multi_search(Context, Type, Props);
+        _Status ->
+            Context1
+    end.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
 -spec validate_query(cb_context:context(), ne_binary()) -> cb_context:context().
--spec validate_query(cb_context:context(), ne_binary(), ne_binaries()) -> cb_context:context().
+-spec validate_query(cb_context:context(), wh_proplist(), wh_proplist() | ne_binary()) -> cb_context:context().
 validate_query(Context, Query) ->
     QueryOptions = query_options(cb_context:account_db(Context)),
-    validate_query(Context, Query, QueryOptions).
+    validate_query(Context, QueryOptions, Query).
 
-validate_query(Context, Query, Available) ->
+validate_query(Context, _Available, []) ->
+    cb_context:set_resp_status(Context, 'success');
+validate_query(Context, Available, [{Query, _}|Props]) ->
+    Context1 = validate_query(Context, Available, Query),
+    case cb_context:resp_status(Context1) of
+        'success' ->
+            validate_query(Context1, Available, Props);
+        _Status ->
+            Context1
+    end;
+validate_query(Context, Available, Query) when is_binary(Query) ->
     case lists:member(Query, Available) of
-        'true' -> cb_context:set_resp_status(Context, 'success');
+        'true' ->
+            cb_context:set_resp_status(Context, 'success');
         'false' ->
             cb_context:add_validation_error(
-              <<"q">>
-              ,<<"enum">>
-              ,wh_json:from_list(
-                 [{<<"message">>, <<"Value not found in enumerated list of values">>}
-                  ,{<<"target">>, Available}
+                <<"query">>
+                ,<<"enum">>
+                ,wh_json:from_list([
```

Comment:
Also, if the query string param is `q`, the validation error should be on `q`. If `q` has been changed to `query`, please make sure docs reflect that.